internal/telemetry/client.go

```diff
@@ -88,21 +99,6 @@ type Client struct {
 	Env     string
 	Version string
 
-	// Determines whether telemetry should actually run.
```

Comment:
Getting rid of configuring this variable and solely relying on reading DD_INSTRUMENTATION_TELEMETRY_ENABLED to decide whether telemetry for simplicity and avoiding conditions where `Disabled` and DD_INSTRUMENTATION_TELEMETRY_ENABLED disagree (e.g. if a client is start/stopped many times and proper cleanup isn't done for resetting `Disabled`). The same goes for other DD_INSTRUMENTATION_TELEMETRY... env vars